flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStIterateOperation.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.forst;
+
+import org.apache.flink.runtime.asyncprocessing.StateRequestType;
+
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * The iterate operation implementation for ForStDB, which leverages rocksdb's iterator directly.
+ */
+public class ForStIterateOperation implements ForStDBOperation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ForStIterateOperation.class);
+
+    public static final int CACHE_SIZE_LIMIT = 128;
+
+    private final RocksDB db;
+
+    private final List<ForStDBIterRequest<?>> batchRequest;
+
+    private final Executor executor;
+
+    ForStIterateOperation(RocksDB db, List<ForStDBIterRequest<?>> batchRequest, Executor executor) {
+        this.db = db;
+        this.batchRequest = batchRequest;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> process() {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+
+        AtomicInteger counter = new AtomicInteger(batchRequest.size());
+        for (int i = 0; i < batchRequest.size(); i++) {
+            ForStDBIterRequest<?> request = batchRequest.get(i);
+            executor.execute(
+                    () -> {
+                        // todo: config read options
+                        try (RocksIterator iter = db.newIterator(request.getColumnFamilyHandle())) {
```

Comment:
This logic seems a bit complex.
Could you add some descriptions in some key steps or split them into methods ?